### DISCUSSION TITLE
Xtouch mini - Knobs working weirdly (pickup/other?) after re-working with the Behringer setup software :-(

### DISCUSSION THREAD

**Discussion by orightok** [rsjaffe/MIDI2LR#756] [Problems]
Hello all, I had my Xtouch mini working fine for few years, but few days ago thought I'd try to play with the mixer channel. 
Ended up not needing to (as ended up not getting an additional mixer as I thought I would), However the whole process brought me to go thru the whole behringer setup software again, and it messed up things. I had this problem few years ago and I just can't find what I did then to fix it:
1. When turning knobs that are set for preferences in the basic tab (temp,tint, exposure, highlights, shadows etc) - sometimes the knob would first need (me) to get to the CURRENT state and would only work from that point and on, and sometimes
2. A little turn of the knob - would bring that preference to the most minimum point, and then would work from there.

Until now, for the past few years, my Midi2LR plugin option of "Pick-up" was turned on. I've tried switching it off/on, and it didn't change anything. I also tried to play with the different states of the knobs in the Behringer software (fan/pan/the rest of them) and also haven't found a way to solve this. 

Any ideas what's goin on in my system and why the knobs not working properly (anymore)?
Thanks,
Ori.

Edit to add: I'm using LR Classic 10.4, with MID2LR 4.3.1.11, and X-touch Mini

**Comment by jayk3368** [comment]
I'm having the same problems too, though I did not play with Behringer's setup software. Last week I bought a new laptop, transferred everything via Apple's Migration Assistant, and updated to MIDI2LR v6, and ever since then, I've been having those two issues. Not sure if it's due to my laptop updating some Behringer driver (maybe similar to you running the setup software), or if it's completely unrelated and instead due to something with the MIDI2LR v6. 

Hopefully this can get resolved!

**Comment by sys3175** [comment]
Maybe related: I keep getting 'controller not synchronized' messages when pickup mode is on, they seem to appear more often when I turn the knobs quickly, but this doesn't always help, and they seem to be random (the controller syncs, you change values, then the message appears again when you turn the knob once more...).  [Version 6.0.1.0/Lightroom Classic 13.1]

**Comment by RBCAMERA** [comment]
I have been experiencing similar problems the last few months. I have done nothing to change my basic setup, Midi2LR used to work flawlessly. Now some of the basic controls appear erratic.  For instance when increasing exposure the slider will jump the wrong way and I have to turn the encoder knob very slowly to get it to respond properly. Same with some other controls. I use an M2 Pro with Sonoma and an X-touch mini.
Any suggestions would be welcome

**Comment by orightok** [comment]
> Hello all, I had my Xtouch mini working fine for few years, but few days ago thought I'd try to play with the mixer channel. Ended up not needing to (as ended up not getting an additional mixer as I thought I would), However the whole process brought me to go thru the whole behringer setup software again, and it messed up things. I had this problem few years ago and I just can't find what I did then to fix it:
> 
>1. When turning knobs that are set for preferences in the basic tab (temp,tint, exposure, highlights, shadows etc) - sometimes the knob would first need (me) to get to the CURRENT state and would only work from that point and on, and sometimes
> 
> 2. A little turn of the knob - would bring that preference to the most minimum point, and then would work from there.
> 
> 
> Until now, for the past few years, my Midi2LR plugin option of "Pick-up" was turned on. I've tried switching it off/on, and it didn't change anything. I also tried to play with the different states of the knobs in the Behringer software (fan/pan/the rest of them) and also haven't found a way to solve this.
> 
> Any ideas what's goin on in my system and why the knobs not working properly (anymore)? Thanks, Ori.
> 
> Edit to add: I'm using LR Classic 10.4, with MID2LR 4.3.1.11, and X-touch Mini

I may have fixed it!
I'm not sure if that's the right/only way(s) but it seems to work. Here is how:

Option 1:
1. Hard reset (factory reset) the Xtouch mini. Which is done by:
a. Unplug it
b. Press both And B buttons together, and while pressing - plug it in.
c. Wait for few secs for A and B lights to slowly flicker. 
d. When only A stays lit - the hard reset is completed and you can let go. The moment you let go, 1 tiny hair light of each of the upper row rotating buttons will light up as well.
2. From there - rerun the plugin from within LR, and proceed your normal LR work as usual.

Option 2 (haven't tried it, might work?):
I'm not too familiar with all the different aspects nor effects of all the different options of the MiniTouch buttons, but when I opened the Xtouch Behringer Editor software to check what changed after the hard reset - I found that the LED RINGs setup changed: they all became set on PAN (earlier I had them on fan and standard and non of those options made a difference..). 
I'm attaching a screen capture of it.

I'm not sure if just this little PAN thing made the whole different, but maybe. So - If someone here is only trying to play with the software(s) and not to a hard reset - do bare in mind, that while playing back and forth between the Behringer Editor Software and MIDI2LR - only one software can communicate at a time with the mixer itself. Which means - if both of the softwares are open at the same time - one might prevent the other one from getting/sending info from/to the mixer. So make sure to close one when you want to check things in the other. Then of course - if it seems like the Behringer Editor software doesn't connect - just unplug the USB cable and replug it for the software to identify it.

I hope this would help someone else too.
Ori.

![Uploading xtouch mini beringer after hard reset.jpg…]()

**Comment by AndyMcFishka** [comment]
Hey! I installed MIDI2LR today and had same problems with knobs. I've debugged lua plugin and found problems in midi values refresh code. There is PR with fix: https://github.com/rsjaffe/MIDI2LR/pull/771

**Reply by RBCAMERA** [reply]
What exactly have you fixed and how do we apply it ?? Does this fix the knobs iratic behavior ??

**Reply by AndyMcFishka** [reply]
There were some logic problems in code that restore knobs values to midi controller. You should wait for @rsjaffe to merge my PR and build a new version of MIDI2LR.
If you don't want to wait - you can check my PR and manually apply changes from https://github.com/rsjaffe/MIDI2LR/pull/771 to your files in adobe lightroom plugins directory by yourself (they are text files and can be edited with any text editor).

Don't forget to write feedback here :)

**Reply by RBCAMERA** [reply]
I installed the new release from Rsjaffe and the jumping slider problem remains un resolved ??

**Reply by AndyMcFishka** [reply]
i don't have mechanical slider on my midi controller, so did not catch that bug. I've fixed only knobs values

**Reply by RBCAMERA** [reply]
Sorry I think I did not make myself clear in my earlier post. When I was talking about sliders I was referring to the sliders in the software. I use an X-Touch Mini with rotary encoders as my midi controller. When increasing the value of say Exposure using the rotary encoder the slider in the software will oft times jump backwards.. I have tried every combo of pick up etc available in Midi2LR but this does not help. I tried resetting and using a newly made profile but the problem persists.
I have copletely cleaned my system of Midi2LR and reinstalled but still have this problem.

**Comment by orightok** [comment]
Have you tried also a factory reset as I mentioned above?

**Reply by RBCAMERA** [reply]
"I tried resetting.".. (That meant factory resetting the X-touch mini..

no dice..

**Reply by RBCAMERA** [reply]
Aha.. I just tried swapping around some of the encoders.. and guess what I think it may be a harware problem.. When I switch to encoders further along the board the problem goes away.. I am going to do some exploratory surrgery and see if theres a board problem..

**Reply by RBCAMERA** [reply]
UPDATE !!  Well I took the thing apart and poured a generous dose of high test Isopropyl Alcohol into the offending encoder and worked the knob back and forth.. rinse and repeat.. re assembled and voila.. it appears to be working again..  Looks like it was just dirty contacts inside.. hmm when was the last time I spilled coffee on my desk....  Anyway I have worked through everything software wise so I am now an expert at all that stuff..   So if anyone else is having a similar problem the IPA bath may be a quick solution..  The X-Touch mini is easy to get apart... six screws.. and the encoder caps are snap on to the pc board so easy to get at the offending twiddly bit and easy to re assemble..

**Comment by orightok** [comment]
Congrats! But the real question is... How long have you been already
thinking about taking it apart, until you decided today to actually go for
it?
:-)

**Reply by RBCAMERA** [reply]
So the problem started late last year.. But I was traveling so not working too much in LR... But I was just dealing with it and hoping for an update.. When nothing improved after the latest release of LR and Midi2LR  I decided to do a deep dive. I did apply the fork that AndyMcFishka documented a couple of weeks ago and then did the full clean and re install.  I was thinking of trying to setup a Virtual Machine and take a look at the Midi setup in the X-Touch software.. and then it just occurred to me that I had not tried swapping assignments to other encoders..   Light Bulb moment..  10 mins later and smiley face...   Sometimes it's the simple things..At least now if anyone else starts to see this it is another route to a possible problem/solution.

**Comment by antzpantz** [comment]
Maybe a troubleshooting tip should be put at the top: clean your
controllers! 😂

**Comment by szega00** [comment]
I am sure cleaning is necessary time to time, but mine was acting up from the second I bought it 3 days ago. I found the problem though. Just go to the x-touch editor and make sure that in the encoder page, the push section has a different channel then the turn section. For example 11 at one and 12 on the lower side of the page( needles to say that you need to change each one). Save it , dump it and ready to go.
